textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/common/KeywordPredicate.java

```diff
@@ -0,0 +1,124 @@
+package edu.uci.ics.textdb.dataflow.common;
+
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import edu.uci.ics.textdb.api.common.Attribute;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.tokenattributes.CharTermAttribute;
+import org.apache.lucene.queryparser.classic.MultiFieldQueryParser;
+import org.apache.lucene.queryparser.classic.ParseException;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import edu.uci.ics.textdb.api.common.IPredicate;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+
+/**
+ *  @author prakul
+ *
+ */
+
+/**
+ * This class handles creation of predicate for querying using Keyword Matcher
+ */
+public class KeywordPredicate implements IPredicate{
+
+    private final List<Attribute> attributeList;
+    private final String[] fields;
+    private final String query;
+    private final Query queryObject;
+    private ArrayList<String> tokens;
+    private Analyzer analyzer;
+
+    public KeywordPredicate(String query, List<Attribute> attributeList, Analyzer analyzer ) throws DataFlowException{
+        try {
+            this.query = query;
+            this.attributeList = attributeList;
+            String[] temp = new String[attributeList.size()];
+
+            for (int i=0;i< attributeList.size();i++){
+                temp[i] = attributeList.get(i).getFieldName();
+            }
+            this.fields = temp;
+            this.tokens = tokenizeQuery(analyzer, this.query);
+            this.analyzer = analyzer;
+            this.queryObject = createQueryObject();
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public boolean satisfy(ITuple tuple) {
+
+        //This method is not used. Implemented for satisfying interface implementation.
```

Comment:
Again, add a "TODO" to take care of in the future?  When will we implement it?  How?